### ISSUE TITLE
Bun crashed when using DuckDB with Full Text Search in weird circumstances

### ISSUE THREAD

**Issue by charnould** [oven-sh/bun#11699]
### How can we reproduce the crash?

For an unknown reason, Bun crashes when using DuckDB with Full Text Search in weird circumstance...
I've created a super minimal reproduction repo: https://github.com/charnould/bug-duckdb-bun

1. Clone repo
2. `bun install`
3. `bun start`

Bun works when `bun start` without modifying code... 
But it crashes when you change `LIMIT` to `2` in `bug.ts`.

If you keep `LIMIT` to `2` in `bug.ts` and comment out all `await` inside `promise.all()` except one or two, it works...  
But it crashed if you uncomment 3+ `await` inside `promise.all()`.

Is it my code? Duck or Bun.
It's a bit blocking.
Thanks.

### Relevant log output

```shell
Bun v1.1.12 (43f0913c) macOS Silicon
Args: "bun" "run" "bug.ts"
Features: jsc tsconfig 
Builtins: "bun:main" "detect-libc" "node:assert" "node:buffer" "node:events" "node:fs" "node:os" "node:path" "node:stream" "node:string_decoder" "node:url" "node:util" "node:util/types" 
Elapsed: 102ms | User: 120ms | Sys: 34ms
RSS: 89.08MB | Peak: 89.08MB | Commit: 1.07GB | Faults: 66

panic(main thread): Segmentation fault at address 0x7AEDBD36283B5934
oh no: Bun has crashed. This indicates a bug in Bun, not your code.

To send a redacted crash report to Bun's team,
please file a GitHub issue using the link below:

 https://bun.report/1.1.12/Mr143f0913AiggggE______207kN+h0nG2487N+mjmFA2sz+296DozstnoB
```

### Stack Trace (bun.report)

Bun v1.1.12 ([`43f0913`](<https://github.com/oven-sh/bun/tree/43f0913c3875cbf260c60868268adc5953b09a80>)) on macos aarch64 [RunCommand]

Segmentation fault at address 0x74DA82DBC29C85EA

- *6 unknown/js code*
- `src.bun.js.event_loop.EventLoop.tickQueueWithCount__anon_362990`
- `src.bun.js.event_loop.EventLoop.tick`
- `src.bun.js.event_loop.EventLoop.waitForPromise`
- `src.bun.js.javascript.OpaqueWrap__anon_55102__struct_286452.callback`

**Comment by charnould** [comment]
After having deleted `await` in `Promise.all()`, it seems to work, but it's not. This is - I think - a bug in `Bun`...

**Comment by Jarred-Sumner** [comment]
@charnould it is 100% a bug in Bun

**Comment by charnould** [comment]
> @charnould it is 100% a bug in Bun

OK, thanks!  
Do you know if it can be fixed soon (I'm totally unable to do it unfortunately)?

**Comment by charnould** [comment]
@Jarred-Sumner is there a way to paid for bug-fix for Bun? It's blocking for a demo I have next week and I'd prefer not to retry Deno or Node :) . Thanks again for making Bun!

**Comment by Jarred-Sumner** [comment]
> @Jarred-Sumner is there a way to pay for bug-fix for Bun? It's blocking for a demo I have next week and I'd prefer not to retry Deno or Node :) . Thanks again for making Bun!

Help us find system engineers in or open to moving to San Francisco and we’ll definitely prioritize your issues.

**Comment by 190n** [comment]
With the fix from #13783, which landed in Bun 1.1.27, your example doesn't crash anymore on my machine. Please let us know if you still experience crashes.

**Comment by charnould** [comment]
> With the fix from #13783, which landed in Bun 1.1.27, your example doesn't crash anymore on my machine. Please let us know if you still experience crashes.

Same! Thanks!
